### ISSUE TITLE
torch.compile timm models passrate went down by one last week, hasn't recovered yet

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#185255]
torch.compile timm models passrate went down by one last week, hasn't recovered yet

<img width="429" height="246" alt="Image" src="https://github.com/user-attachments/assets/9abdb674-729d-402a-8643-1770ab893dae" />

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by zou3519** [comment]
Hmm I can't reproduce this locally. Basically the failing model appears to fail for all recent commits on my machine. claude thinks the repro command is the following: `python benchmarks/dynamo/timm_models.py --device cuda --accuracy --amp \
  --training --inductor --disable-cudagraphs --only vit_base_patch16_siglip_256`

claude says:
```
Bisect: vit_base_patch16_siglip_256 accuracy failure in timm_models training/amp

The model vit_base_patch16_siglip_256 started failing accuracy on the compile dashboard between
commits 89afe5d7692a (May 19, passing) and 43f53ca119d9 (May 21, fail_accuracy). This is a
training/amp/h100 regression that affects all inductor backends (cudagraphs, default,
cpp_wrapper, etc.). Two other models (inception_v3, mobilenetv2_100) were already failing before
this — vit_base_patch16_siglip_256 is the new one.
```

**Comment by zou3519** [comment]
backend="inductor" fails for me, eager/aot_eager pass

**Comment by zou3519** [comment]
I did a bisection and https://github.com/pytorch/pytorch/pull/182897 made the error on this increase
```
Found the RMSE jump culprit:

b2bc2b83bcee4495b14c3346bf97e952432b27db
[inductor] Lower nested reductions in SIMD codegen (#182897)

Bisect criterion was RMSE (res-fp64) > 0.45 as bad:

- Before: 705aa621264 -> 0.34708
- First bad: b2bc2b83bce -> 0.57664
- Reported bad end 43f53ca119d9 -> 0.57664

The backend check on 43f53ca119d9 also narrows it:

- --backend eager -> pass
- --backend aot_eager -> pass
- --inductor -> RMSE 0.57664, fail_accuracy

Sanity check on the first bad commit with the new feature disabled:

--inductor-config triton.nested_reduction=False

drops back to RMSE 0.34708, so the higher divergence is specifically tied to the nested reduction
lowering enabled by b2bc2b83bce.

Logs are under agent_space/bisect_logs/. The checkout is currently left at the bisect result
commit.
```

I'm going to try a revert first (revert first, ask questions later) and see if the CI goes back to green.. if it doesn't then I'll keep digging.

**Comment by eellison** [comment]
@zou3519 the only thing that changed was reduction order. i will submit a pr to fix the assertions.

**Comment by zou3519** [comment]
thank you

**Comment by jansel** [comment]
My agent was unable to reproduce this on 9d625a9bfcf51383132bb6dad90421d2459f5cff. Reopen if you see this on the latest PyTorch nightly
